### DISCUSSION TITLE
Libbi Target Charge

### DISCUSSION THREAD

**Discussion by MalcolmSpencer** [CJNE/ha-myenergi#567] [General]
I have a Libbi battery size of 10.2 Kwh but when I set the target charge to 100% in the App the target charge in HA is showing as 9.2 Kwh (50% = 4.6 Kwh) any thoughts on why this might be and is everyone else seeing the same?

@trizmark @jwillemsen

**Comment by trizmark** [comment]
It's normal. The libbi has a depth of discharge setting and it won't discharge below that percentage. The reported SoC and target charge already accounts for this, so 100% is less than your total battery capacity. My 20.4kWh system is showing 18.4kWh as maximum.

**Reply by MalcolmSpencer** [reply]
thanks for that so if i want 100% I presume I should set the target to 9.2kWh but I know it rarely reaches 100% and usually stops at 98%

**Reply by trizmark** [reply]
Yes, indeed, set it to 9.2kW. I think the API accepts values up to the actual capacity of the battery.

**Reply by MalcolmSpencer** [reply]
Thanks and thanks for all the effort that has gone into creating this integration

**Comment by G6EJD** [comment]
It’s all normal and as stated the depth of discharge will never be 100% and you’ll waste more energy at least 20% to charge up the battery and then another 10-15% to discharge the battery, it’s not a perfect world never will be.

**Comment by G6EJD** [comment]
It’s all about protecting the battery, when charged above 80% that’s when battery damage occurs, due to heating effects, as the electro-chemical reaction slows and heat is generated, then there is a life reduction and plate damage and so-on, so the battery management system will charge in three distinct ways, initially at constant current, then constant voltage and finally a balancing charge where each cell is brought up to the same voltage, so at 98% the BMS will most likely be trying to balance the cells, which can take many hours, so it may eventually get to 100%.

anyway how much energy is in 2%? it’s 10.2 x 0.02 that’s 0.2kWh it’s of no great value to most situations.
